pyflowater/flostream.py

```diff
@@ -0,0 +1,56 @@
+"""Code relating to the firestore API Flo use to stream live data."""
+
+import requests
+import json
+from google.cloud import firestore
+from google.oauth2.credentials import Credentials
+
+from pyflowater.const import (
+    FLO_GOOGLE_API_KEY,
+    FIREBASE_REST_API,
+    FLO_FIRESTORE_PROJECT,
+)
+
+def _get_token_info(token):
+    url = f"{FIREBASE_REST_API}/verifyCustomToken?key={FLO_GOOGLE_API_KEY}"
+    headers = {"Content-type": "application/json; charset=UTF-8"}
+    data = json.dumps({"returnSecureToken": True, "token": token})
+    resp = requests.post(url, headers=headers, data=data)
+    return resp.json()
+
+class FloListener:
+    """Flo firestore listener class."""
+
+    def __init__(self, token, deviceId, callback):
+        self._token = token
+        self._deviceId = deviceId
+        self._callback = callback
+        self._watch = None
+        self._client = None
+        self._doc_ref = None
+
+    def set_callback(self, callback):
+        """Update the callback."""
+        self._callback = callback
+
+    def start(self):
+        """Begin listening on the firestore database."""
```

Comment:
Please mention here that the callback will be called almost immediately to supply the initial document. Also mention that server-side resources are consumed on every call to the callback function, including the first call. So don't stop() if you're going to quickly start() again.

Reply:
+1  Great point.